engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java

```diff
@@ -640,6 +640,19 @@ protected NetworkOrchestrator() {
         setStateMachine();
     }
 
+    private void updateNetworkDetails(NetworkVO networkPersisted, Network network) {
+        NetworkDetailVO networkDetailVO = null;
+        if (isNotBlank(network.getRouterIp())) {
+            networkDetailVO = new NetworkDetailVO(networkPersisted.getId(), ApiConstants.ROUTER_IP, network.getRouterIp().toString(), true);
+        }
+        if (isNotBlank(network.getRouterIpv6())) {
+            networkDetailVO = new NetworkDetailVO(networkPersisted.getId(), ApiConstants.ROUTER_IPV6, network.getRouterIpv6().toString(), true);
```

Comment:
this will override the above _networkDetailVO_ , as per the code when both are not blank and always the latest one is persisted. Either keep if-else for IPv4 & IPv6, or move the network details persist to respective if block.

Reply:
But based on the code for persisting placeholder record, it seems like one can specify both ipv4 and ipv6 details:
https://github.com/shapeblue/cloudstack-kddi/blob/kddi-release/server/src/com/cloud/network/router/NetworkHelperImpl.java#L701
(and L718)